keps/0019-package-api-versioning.md

```diff
@@ -0,0 +1,107 @@
+---
+kep-number: 19
+title: Versioning of Operator Packages
+authors:
+  - "@nfnt"
+owners:
+  - "@nfnt"
+creation-date: 2019-11-05
+last-updated: 2019-11-05
+status: provisional
+---
+
+# Versioning of Operator Packages
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Support multiple API versions in the package repository](#support-multiple-api-versions-in-the-package-repository)
+  * [Install a specific application version](#install-a-specific-application-version)
+  * [Naming of package tarballs](#naming-of-package-tarballs)
+  * [Semantic versioning](#semantic-versioning)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+The different versions of components that describe an operator package are surfaced to users, so that they are able to install exactly specify which version can be installed. Furthermore, compatibility with existing packages for older APIs of KUDO is provided.
+
+## Motivation
+
+The dependencies between KUDO, operators packages and the application that is managed by the operator enforce the use of multiple versions to fully describe these dependencies. First, there is the dependency on a specific API version of KUDO. Also, a specific version of an application is provided by the package. And last, the operator itself may have multiple revisions. All of these versions need to be captured as part of the operator in a concise way that makes it easy for users to install specific operators for any of these versions.
```

Comment:
this motive misses the fact that users of an underlying tech usually care about that version.   Which adds another dimension to the problem.

Reply:
when we add the details of underlying tech version... we also split the categories of concerns into operators that have an underlying tech version and those that do not.  both will need a solution defined here.

Reply:
Thanks, let me add some sentences for this problem. I'll also mention the "bundling problem" discussed below.